core/database/foxx/api/user_router.js

```diff
@@ -516,9 +519,54 @@ router.get('/token/set', function(req, res) {
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false
-                    });
+                    if (token_type && other_token_data) {
+                        // We need the '_key' from the user document, which is not necessarily a part of the above checks.
+                        const user_doc = g_db.u.document(user_id);
+
+                        // find or insert collection
+                        const collection_search_key = other_token_data;
```

Comment:
An important note here, you are assuming this just the UUID. Other token data is a string which is not guaranteed to just be the uuid.

You will need a function here to verify the contents. For instance, it may also contain the scopes.

You might ask why not include this as a separate query parameter? My thought behind this is if we support tokens to other applications or services they may not have a scope option. 

We need to have a discussion around what the other_token_data query parameter should and can contain, if it is a globus token. Right now, it should contain scopes and the uuid. What the format is also needs to be defined.

Perhaps?

"<uuid>:<scopes>"

You could then have a JavaScript function that parses the string. 
```
const { uuid, scopes } = parseGlobusTokenData( other_token_data );
```

Reply:
If we go this way, we will have to also be sure the core service knows how to correctly pass the globus token data as well.